components/automate-ui/src/app/components/resource-dropdown/resource-dropdown.component.ts

```diff
@@ -100,14 +110,33 @@ export class ResourceDropdownComponent implements OnInit, OnChanges {
 
   closeDropdown(): void {
     this.dropdownState = 'closed';
-    this.onDropdownClosing.emit(
-      this.resources.filter(r => r.checked).map(r => r.id));
+    const flattenedIDs = [].concat(
+      ...this.snapshotResources.map(
+        resource => resource.itemList.filter(r => r.checked).map(r => r.id)));
+    this.onDropdownClosing.emit(flattenedIDs);
   }
 
   handleFilterKeyUp(): void {
-    this.filteredResources = this.resourcesInOrder.filter(resource =>
-      resource.id.toLowerCase().indexOf(this.filterValue.toLowerCase()) > -1
-    );
+    for (let i = 0; i < this.filteredResources.length; i++) {
+      this.filteredResources[i].itemList =
+        this.snapshotResources[i].itemList.filter(r =>
+          r.name.toLowerCase().indexOf(this.filterValue.toLowerCase()) > -1);
```

Comment:
Hmm so we're testing if it's a (case-insensitive) substring? 💡

Reply:
matching case-insensitively, yes.